inc/shortcodes/class-shortcode.php

```diff
@@ -55,4 +55,54 @@ public static function callback( $attrs, $content = '' ) {
 		return '';
 	}
 
+	/**
+	 * Parse iframes from a string, if there are any
+	 *
+	 * @param string $content
+	 * @return array|false
+	 */
+	protected static function parse_iframes( $content ) {
+
+		if ( false === stripos( $content, '<iframe' ) ) {
+			return false;
+		}
+
+		if ( preg_match_all( '#(.+\r?\n?)?(<iframe([^>]+)>[^<]{0,}</iframe>)(\r?\n?.+)?#', $content, $matches ) ) {
```

Comment:
I'm not sure what you're trying to do with the beginning and end delimiters. `(.+\r?\n?)?` Is that just supposed to swallow the previous and next lines of the content if any exist?